configure

```diff
@@ -292,277 +332,749 @@ LIBS="${LIBS} mpfr"
+                    BUILD_MACHINE="x86"
+                    ;;
+                silvermont-*)
+                    MARCH_FLAG="-march=silvermont"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                goldmont-*)
+                    MARCH_FLAG="-march=goldmont -march=silvermont"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                goldmont-plus-*)
+                    MARCH_FLAG="-march=goldmont-plus -march=goldmont"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                tremont-*)
+                    MARCH_FLAG="-march=tremont -march=goldmont"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                core2-*)
+                    MARCH_FLAG="-march=core2"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                corei-* | coreinhm-* | nehalem-*)
+                    MARCH_FLAG="-march=nehalem -march=core2"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                coreiwsm-* | westmere-*)
+                    MARCH_FLAG="-march=westmere -march=nehalem"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                coreisbr-* | coreisbrnoavx-* | sandybridge-* |\
+                sandybridgenoavx-*)
+                    MARCH_FLAG="-march=sandybridge -march=westmere"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                coreiibr-* | coreiibrnoavx-* | ivybridge-* | ivybridgenoavx-*)
+                    MARCH_FLAG="-march=ivybridge -march=sandybridge"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                coreihwl-* | coreihwlnoavx-* | haswell-* | haswellnoavx-*)
+                    MARCH_FLAG="-march=haswell -march=ivybridge"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                coreibwl-* | coreibwlnoavx-* | broadwell-* | broadwellnoavx-*)
+                    MARCH_FLAG="-march=broadwell -march=haswell"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                skylake-avx512-*)
+                    MARCH_FLAG="-march=skylake -march=broadwell -mno-avx512f"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                skylake-* | skylakenoavx-* | kabylake-* | kabylakenoavx-* |\
+                coffeelake-* | coffeelakenoavx-*)
+                    MARCH_FLAG="-march=skylake -march=broadwell"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                knl*-*)
+                    MARCH_FLAG="-march=knl -mno-avx512f"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                knm*-*)
+                    MARCH_FLAG="-march=knm -mno-avx512f"
+                    BUILD_MACHINE="x86_64"
+                    ;;
+                cascadelake-*)
+                    MARCH_FLAG="-march=cascadelake -march=skylake -mno-avx512f"
```

Comment:
Interesting. Doesn't the second `-march` value overwrite the first? Is there a special behavior for multiple such options? I looked at the GCC docs but saw nothing about this, but maybe I looked in the wrong spot?

Reply:
Looking at GMP, they specify it in a falling order such that the first one is the most "modern" flag and the last one is the most "outdated" flag. I haven't seen it anywhere else, but it compiles on my system.

Reply:
Are you sure GMP actually supplies all of them together and that you aren't looking at a list which it tries one at a time in its configure script?